### PR TITLE
docs(forms): fix broken link to Forms package in API reference

### LINKED ISSUE #56846: DOCUMENTATION - Link to https://angular.dev/api/forms broken
### Describe the problem that you experienced

https://angular.dev/guide/forms/reactive-forms#reactive-forms-api-summary

there's a link here to the API ref for [Forms package](https://angular.dev/api/forms), which is broken

### Enter the URL of the topic with the problem

https://angular.dev/guide/forms/reactive-forms#reactive-forms-api-summary

### Describe what you were looking for in the documentation

FormArray API reference

### Describe the actions that led you to experience the problem

_No response_

### Describe what you want to experience that would fix the problem

_No response_

### Add a screenshot if that helps illustrate the problem

_No response_

### If this problem caused an exception or error, please paste it here

_No response_

### If the problem is browser-specific, please specify the device, OS, browser, and version

_No response_

### Provide any additional information here in as much as detail as you can

_No response_

### PR DESCRIPTION
**Pull request by ashishnd**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

https://angular.dev/guide/forms/reactive-forms#reactive-forms-api-summary

there's a link here to the API ref for [Forms package](https://angular.dev/api/forms), which is broken

Issue Number: Fixes #56846 

## What is the new behavior?
The link to the API reference for the Forms package now correctly points to [Forms package](https://angular.dev/api#forms).

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by ashishnd** [comment]
Hey @alxhub, @JeanMeche! This is my first contribution and I did the necessary changes to the code, but I used my new email id for signing the cla and the one associated with my github account was my old email id. 

I did update github with my new email id once I noticed I was still using the old one, but I did this after submitting the pr and now the authors list has my old email and hence it's failing the cla/google. 

I tried to update the authors in a subsequent commit and somehow managed to add my new mail to the list of authors but I couldn't remove the old one and it's still failing the cla. Can you please guide me with what I should do in order to fix it? 

I'm really sorry for the inconvenience!

File: adev/src/content/guide/forms/reactive-forms.md [hunk 1]
```diff
@@ -395,7 +395,7 @@ Initially, the form contains one `Alias` field. To add another field, click the
 ## Reactive forms API summary
 
 The following table lists the base classes and services used to create and manage reactive form controls.
-For complete syntax details, see the API reference documentation for the [Forms package](api/forms "API reference").
+For complete syntax details, see the API reference documentation for the [Forms package](https://angular.dev/api#forms "API reference").
```
Comment (JeanMeche) [inline adev/src/content/guide/forms/reactive-forms.md:398]: Can you please make this a relative link ? So this link also work on https://next.angular.dev. 
```suggestion
For complete syntax details, see the API reference documentation for the [Forms package](/api#forms "API reference").
```

**Comment by JeanMeche** [comment]
Also can you please squash your commits into a single one, we'd like this change to be a single commit.

**Comment by ashishnd** [comment]
1. I have made the link relative so that it also works on https://next.angular.dev/ 
2. I've squashed the commits into a single commit.
@JeanMeche, please let me know if any other changes are to be made?
Sorry for the inconvenience!

**Comment by JeanMeche** [comment]
Can you also have a look at that confligt the PR is reporting ?

File: adev/src/content/guide/forms/reactive-forms.md [hunk 2]
```diff
@@ -395,7 +395,7 @@ Initially, the form contains one `Alias` field. To add another field, click the
 ## Reactive forms API summary
 
 The following table lists the base classes and services used to create and manage reactive form controls.
-For complete syntax details, see the API reference documentation for the [Forms package](api#forms "API reference").
+For complete syntax details, see the API reference documentation for the [Forms package](/api#forms "API reference" "API reference").
```
Comment (JeanMeche) [inline adev/src/content/guide/forms/reactive-forms.md:398]:
```suggestion
For complete syntax details, see the API reference documentation for the [Forms package](/api#forms "API reference").
```

Comment (JeanMeche) [reply adev/src/content/guide/forms/reactive-forms.md:398]: Also can you drop that merge commit, thank you.

**Comment by ashishnd** [comment]
Hey @JeanMeche, I am actually struggling to squash the 2 commits into a single one. 

Although I did it yesterday for another issue here they just wont get squashed into a single commit when i try to do so using rebase. It says "commit <commit hash> is a merge but no -m option was given". 

Can you please nudge me in the correct direction by telling me what to do here? 

Apologies for the inconvenience.

**Comment by JeanMeche** [comment]
What I would do in your situation: 
* `git rebase -i HEAD~2` and drop the merge commit
* update your origin to match upstream (you can do this from your own fork https://github.com/ashishnd/angular) 
* fetch origin to update main to the latest
* `git rebase main`

And you should be fine.

**Comment by JeanMeche** [comment]
Actualy the particular issue was fixed by #56848, this is why you probably have an issue without your change.

**Comment by ashishnd** [comment]
So these changes not working out was not my fault right? I was trying to figure out the problem for hours because I thought since I am not very good at git maybe I was doing something wrong. Thanks for informing!

**Pull request closed without merging**